Automattic/simplenote-android
# Add unverified account check login response

Fixes #1445 

### Fix

This PR adds support to handle account verification responses to login requests. If the account is unverified, a `403` response is sent, therefore, blocking the user from login. For `403` responses, a dialog is shown that the user has an unverified account and a button that the user can tap to send an email to verify their account. 

![20210823_091125](https://user-images.githubusercontent.com/195721/130462352-e3567574-523f-481f-a9d8-de71dc498017.gif)


### Test
1. Tap ***Log In*** button.
2. Tap ***Log In with Email*** button.
3. Enter email address in ***Email*** field.
4. Notice ***Log In*** button is disabled.
5. Enter incorrect password of more than four and less than eight characters in ***Password*** field.
6. Notice ***Log In*** button is enabled after four characters are input.
7. Tap ***Log In*** button.
8. Notice ***Error*** dialog that shows invalid credentials.
9. Tap ***OK*** button in dialog.
10. Enter correct password of more than seven characters in ***Password*** field.
11. Notice ***Log In*** button is enabled after four characters are input.
12. Tap ***Log In*** button.
13. Notice app is successfully logged in.
14. Logout and know try a user with unverified account. If the backend implementation has not been implemented, go to line `44-48` and interchange `INVALID_ACCOUNT` BY `UNVERIFIED_ACCOUNT` (this will show the unverified account dialog for a invalid credentials response):
```java
 case INVALID_ACCOUNT:
    showUnverifiedAccountDialog();
    break;
case UNVERIFIED_ACCOUNT:
default:
```
14. Go to login and type the correct credentials of the account with a compromised password or invalid password if it has not been implemented.
15. Notice ***Error*** dialog that shows unverified account. 
16. Tap on the `Resend Verification Email`
17. You must have received an email to verify your account. This email is rate limited, thus, you should be able to send just one email per day to the same email.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in 79f24220f47b4e5d780db1fde883958c95ee87c8 with:
> [Internal] Added login alert for unverified accounts
